src/shared/logs-show.c

```diff
@@ -1073,7 +1073,9 @@ int add_matches_for_unit(sd_journal *j, const char *unit) {
         );
 
         if (r == 0 && endswith(unit, ".slice")) {
-                char *m5 = strappend("_SYSTEMD_SLICE=", unit);
+                char *m5;
+
+                m5 = strjoina("_SYSTEMD_SLICE=", unit);
 
                 /* Show all messages belonging to a slice */
                 (void)(
```

Comment:
While at it, while not change this to:
```
r = sd_journal_add_disjunction(j);
if (r >= 0)
    r = sd_journal_add_match(j, m5, 0);
```
Or is this code doing something unique that I'm not seeing?